tree/master/cloud/src/solution/Microsoft.Legal.MatterCenter.Service/Microsoft.Legal.MatterCenter.Repository/Matter/MatterRepository.cs

```diff
@@ -1514,24 +1519,193 @@ internal string GetStampPropertyValue(Dictionary<string, object> stampedProperty
+                            int optionCounter = 1;
+
+                            foreach (string option in options)
+                            {
+                                jw.WriteStartObject();
+                                jw.WritePropertyName("choiceId");
+                                jw.WriteValue(optionCounter);
+                                jw.WritePropertyName("choiceValue");
+                                jw.WriteValue(option);
+                                optionCounter++;
+                                jw.WriteEndObject();
+                            }
+                            jw.WriteEndArray();
+                        }
+                        else if (field.TypeAsString == "MultiChoice")
+                        {
+                            jw.WritePropertyName("type");
+                            jw.WriteValue(Convert.ToString(((Microsoft.SharePoint.Client.FieldMultiChoice)field).TypeAsString));
+                            List<string> options = GetChoiceFieldValues(clientContext, field);
+                            jw.WritePropertyName("values");
+                            jw.WriteStartArray();
+                            int optionCounter = 1;
+
+                            foreach (string option in options)
+                            {
+                                jw.WriteStartObject();
+                                jw.WritePropertyName("choiceId");
+                                jw.WriteValue(optionCounter);
+                                jw.WritePropertyName("choiceValue");
+                                jw.WriteValue(option);
+                                optionCounter++;
+                                jw.WriteEndObject();
+                            }
+
+                            jw.WriteEndArray();
+                        }
+                        else
+                        {
+                            jw.WritePropertyName("type");
+                            jw.WriteValue(Convert.ToString(field.TypeAsString));
+                        }
+                        jw.WriteEndObject();
+                    }
+                }
+                jw.WriteEndArray();
+                jw.WriteEndObject();
+                return sb.ToString();
+            }
+            catch (Exception ex)
+            {
+                customLogger.LogError(ex, MethodBase.GetCurrentMethod().DeclaringType.Name, MethodBase.GetCurrentMethod().Name, logTables.SPOLogTable);
+                throw;
+            }
+        }
+        /// <summary>
+        /// This method is to get the choice field values of choice data type
+        /// </summary>
+        /// <param name="clientContext"></param>
+        /// <param name="fieldName"></param>
+        /// <returns></returns>
+        private static List<string> GetChoiceFieldValues(ClientContext clientContext, SharePoint.Client.Field fieldName)
+        {
+            List<string> fieldList = new List<string>();

```

Comment:
Purpose: To retrieve and populate options for choice type field and create json object.